doc/features/speculative-executions/README.md

```diff
@@ -0,0 +1,214 @@
+# Speculative query execution
```

Comment:
One thing was just thinking about while reading this is that these concepts are not specific to any driver per se.  I.E. we have mostly duplicated documentation between the C#, java and nodejs drivers with regards to speculative execution and query idempotence.   Wonder if going forward we could document these concepts at a higher level so we don't have to repeat ourselves amongst the docs, but not really sure if that is worth it.

Reply:
Its true that is 98% alike and its weird to maintain "copies", but the code samples and details (like further support of certain config settings) I think justifies for different doc sources.